google/cloud/storage/internal/signed_url_requests.cc

```diff
@@ -80,6 +82,132 @@ std::ostream& operator<<(std::ostream& os, V2SignUrlRequest const& r) {
   return os << "SingUrlRequest={" << r.StringToSign() << "}";
 }
 
+namespace {
+std::string QueryStringFromParameters(
+    CurlHandle& curl,
+    std::multimap<std::string, std::string> const& parameters) {
+  std::string result;
+  char const* sep = "";
+  for (auto const& qp : parameters) {
+    result += sep;
+    result += curl.MakeEscapedString(qp.first).get();
+    result += '=';
+    result += curl.MakeEscapedString(qp.second).get();
+    sep = "&";
+  }
+  return result;
+}
+
+std::string TrimHeaderValue(std::string const& value) {
+  std::string tmp = value;
+  tmp.erase(0, tmp.find_first_not_of(' '));
+  tmp = tmp.substr(0, tmp.find_last_not_of(' ') + 1);
```

Comment:
Nit: It is a little funny to even try to compute (npos + 1), but I'll grant that would imply tmp.empty() at which point it doesn't matter what the second parameter to substr() is [and everything is unsigned].

Reply:
Left it as-is, but if you have a suggestion on how to make this less convoluted / more readable I would be glad to hear it, I cannot say that I like the current code that much.